### PR TITLE
[639] Add how to use init system conf guide

### PR DESCRIPTION
**Pull request by rina23q**
## Proposed changes

The doc to explain user how to give `system.toml` to configure their preferred init system working with thin-edge.io.
#639 is prior to this PR.

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: docs/src/SUMMARY.md [hunk 1]
```diff
@@ -32,6 +32,7 @@
     - [How to restart your thin-edge.io device](./howto-guides/016_restart_device_operation.md)
     - [How to use apama software management plugin](./howto-guides/017_apama_software_management_plugin.md)
     - [How to change temp path](./howto-guides/018_change_temp_path.md)
+    - [How to use thin-edge.io with your preferred init system](./howto-guides/019_how_to_use_preferred_service_manager.md)
```
Comment (itsyitsy) [inline docs/src/SUMMARY.md:35]:
```suggestion
    - [How to use thin-edge.io with your preferred init system](./howto-guides/019_how_to_use_preferred_init_system.md)
```
Why should that be different for title and file name? Of course also apply for actual file name

File: docs/src/howto-guides/019_how_to_use_preferred_service_manager.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# How to use thin-edge.io with your preferred init system
+
+thin-edge.io works seamlessly with `systemd` on the CLI commands `tedge connect` and `tedge disconnect`.
+However, not all OS support `systemd`.
+You might want to use other init systems like `OpenRC`, `BSD`, `init.d` with thin-edge.io.
```
Comment (itsyitsy) [inline docs/src/howto-guides/019_how_to_use_preferred_service_manager.md:5]:
```suggestion
You might want to use another init system like `OpenRC`, `BSD`, `init.d` with thin-edge.io.
```

File: docs/src/howto-guides/019_how_to_use_preferred_service_manager.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+is_available = ["/sbin/rc-service", "-l"]
+restart = ["/sbin/rc-service", "{}", "restart"]
+stop =  ["/sbin/rc-service", "{}", "stop"]
+enable =  ["/sbin/rc-update", "add", "{}"]
+disable =  ["/sbin/rc-update", "delete", "{}"]
+is_active = ["/sbin/rc-service", "{}", "status"]
+```
+
+Then, adjust the values with your preferred init system.
+To get to know the rules of the configuration file, please refer to [Init System Configuration File](./../references/system-config.md).
```
Comment (itsyitsy) [inline docs/src/howto-guides/019_how_to_use_preferred_service_manager.md:33]: Probably we should also update that linked files name to "init-system-config.md". That way its more clear that it is only about the init system.

File: docs/src/howto-guides/019_how_to_use_preferred_service_manager.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+After you finish creating your own configuration file, it's good to limit the file permission to read-only.
+
+```shell
+sudo chmod 444 /etc/tedge/system.toml
+```
+
+Now `tedge connect` and `tedge disconnect` use the init system that you specified!
```
Comment (itsyitsy) [inline docs/src/howto-guides/019_how_to_use_preferred_service_manager.md:41]:
```suggestion
Now `tedge connect` and `tedge disconnect` will use the init system that you specified!
```

File: docs/src/howto-guides/019_how_to_use_preferred_service_manager.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+## Default settings
+
+If the custom configuration file `/etc/tedge/system.toml` is not placed,
```
Comment (itsyitsy) [inline docs/src/howto-guides/019_how_to_use_preferred_service_manager.md:45]:
```suggestion
If the custom configuration file `/etc/tedge/system.toml` is not in place,
```

**Review by itsyitsy — CHANGES_REQUESTED** [review]

File: docs/src/howto-guides/019_how_to_use_preferred_service_manager.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
+# How to use thin-edge.io with your preferred init system
+
+thin-edge.io works seamlessly with `systemd` on the CLI commands `tedge connect` and `tedge disconnect`.
+However, not all OS support `systemd`.
+You might want to use other init systems like `OpenRC`, `BSD`, `init.d` with thin-edge.io.
+For this demand, this guide explains how to give a custom init system configuration to thin-edge.io.
```
Comment (didier-wenzek) [inline docs/src/howto-guides/019_how_to_use_preferred_service_manager.md:6]: I would be more direct.

```suggestion
This guide explains how to configure thin-edge.io to use another init system.
```

File: docs/src/SUMMARY.md (hunk 1 above)
Comment (rina23q) [reply docs/src/SUMMARY.md:35]: Good catch!

File: docs/src/howto-guides/019_how_to_use_preferred_service_manager.md (hunk 3 above)
Comment (rina23q) [reply docs/src/howto-guides/019_how_to_use_preferred_service_manager.md:33]: I think you changed the file name, and my PR used the old name :D

**Review by itsyitsy — APPROVED** [review]

**Pull request merged**